llama_stack/distribution/server/auth_providers.py

```diff
@@ -213,6 +216,12 @@ class OAuth2TokenAuthProviderConfig(BaseModel):
             "namespace": "namespaces",
         },
     )
+    # options specific to introspection:
+    introspection_url: str | None = None
```

Comment:
I think we could bundle up the subfields a bit better to make things clearer:

```
  jwks: OAuth2JWKSConfig | None
  introspection: OAuth2IntrospectionConfig | None
```

all of the `client_id`, `client_secret` etc. can now be folded within the introspection config?

Reply:
I've made that change now.